### ISSUE TITLE
Inability to run pure evaluation task from checkpoint / Hydra overrides for max_steps ignored when resuming with different batch size Description

### ISSUE THREAD

**Issue by FaboNo** [simchowitzlabpublic/nano-world-model#14]
Hi,

First of all, thank you for sharing the code for this World Model project.

I am currently trying to run the evaluation task on the PushT dataset using a local machine (single RTX 5090, 32GB VRAM). Due to VRAM limitations during the evaluation phase (the VAE + metric loggers easily saturate the memory), I had to scale down the batch_size from 40 to 8, and eventually to 1 for evaluation.

However, I am facing major bottlenecks with PyTorch Lightning and Hydra configuration overrides when attempting to resume from a checkpoint just to compute metrics (FID/FVD/LPIPS).

Issues encountered:
trainer.fit is hardcoded / Pure evaluation task is ignored:
Even when passing ++experiment.task=evaluate or changing the task to evaluate inside the YAML file, the script completely ignores it and systematically falls back to a training loop (trainer.fit).

Hydra overrides for max_steps are ignored upon resuming:
I managed to train a model up to 90,000 steps with a batch size of 40. When attempting to resume the run from this checkpoint to finish the remaining 10,000 steps using a safer batch size of 8, PyTorch Lightning recalculates the epoch length based on the new batch size (showing 248,703 steps for Epoch 0).

Attempts to force an early stop or target a specific step via CLI overrides like ++trainer.max_steps=100000 or ++experiment.trainer.max_steps=16000 are completely ignored by Hydra. The framework resets the session step counter to 0 and blindly launches a massive training loop.

Inconsistent dataset fallback on failed overrides:
When trying to aggressively override configuration nodes via the CLI to skip training (e.g., setting max_steps=0), Hydra silently switches the configuration context. For instance, while explicitly requesting experiment=dino_wm_pusht, the script randomly attempted to look for the point_maze dataset paths, resulting in a FileNotFoundError for states.pth.

Questions:
What is the official, intended way to pass a .ckpt file to the script to trigger only the trainer.validate() or trainer.test() pipeline without triggering a new training factory loop?

Why are the trainer.max_steps CLI overrides bypassed when resuming an experience via +trainer.resume_from_checkpoint?

Any guidance on how to properly evaluate a pre-trained checkpoint without hacking train_experiment.py directly would be greatly appreciated.

Environment:

OS: Pop!_OS (Ubuntu-based)

GPU: NVIDIA RTX 5090 (32GB)

Thank you very much for answering my questions

**Comment by knightnemo** [comment]
Thank you for using our repo.
> trainer.fit is hardcoded / Pure evaluation task is ignored:
Even when passing ++experiment.task=evaluate or changing the task to evaluate inside the YAML file, the script completely ignores it and systematically falls back to a training loop (trainer.fit).

The evaluation entry point is controlled by `experiment.tasks`, not `experiment.task`. The intended pure-evaluation command is:

```sh
python src/main.py experiment=evaluate_only \
      dataset=dino_wm/pusht model=nanowm_b2 \
      experiment.resume_from_checkpoint=/path/to/checkpoint.ckpt
```

`experiment=evaluate_only` sets `experiment.tasks=[evaluate]`, so the experiment dispatcher calls `TrainExperiment.evaluate()`, which runs `trainer.validate(...)` only and does not call `trainer.fit(...)`. Using `++experiment.task=evaluate` creates a new unused config key, so it will not affect dispatch. I have updated the code/docs to make this clearer and to use `experiment.resume_from_checkpoint` consistently.

> Hydra overrides for max_steps are ignored upon resuming:
I managed to train a model up to 90,000 steps with a batch size of 40. When attempting to resume the run from this checkpoint to finish the remaining 10,000 steps using a safer batch size of 8, PyTorch Lightning recalculates the epoch length based on the new batch size (showing 248,703 steps for Epoch 0).

You are right. This was a bug / unclear behavior in our resume path. Previously, `experiment.resume_from_checkpoint` did not perform a full PyTorch Lightning resume. We were manually loading checkpoint weights into the module with `load_state_dict(...)`, then calling `trainer.fit(...)` without passing `ckpt_path`. That restores model weights, but not optimizer state, scheduler state, epoch, or `global_step`.

This has now been fixed: during training, `experiment.resume_from_checkpoint=/path/to/ckpt` is passed to Lightning as `trainer.fit(..., ckpt_path=...)`, so optimizer/scheduler/global_step are restored.

To continue a 90k run to 100k with a smaller batch size, use:
```sh
python src/main.py experiment=dino_wm_pusht dataset=dino_wm/pusht model=nanowm_b2 \
    experiment.resume_from_checkpoint=/path/to/90k.ckpt \
    experiment.training.max_steps=100000 \
    experiment.training.batch_size=8
```

> Attempts to force an early stop or target a specific step via CLI overrides like `++trainer.max_steps=100000` or `++experiment.trainer.max_steps=16000` are completely ignored by Hydra. The framework resets
  the session step counter to 0 and blindly launches a massive training loop.

This is related to the same resume issue, plus an override-path issue.

The trainer max step is configured at:

```sh
experiment.training.max_steps=100000
```
not:
```
trainer.max_steps=100000
experiment.trainer.max_steps=100000
```
Overrides like ++trainer.max_steps=... create new unused config keys, so Hydra may accept them, but the training code never reads them. Previously, even with the correct experiment.training.max_steps
override, experiment.resume_from_checkpoint only loaded model weights and did not restore Lightning trainer state, so global_step restarted from 0. That made max_steps behave like a fresh-run step budget.

With the fix above, global_step is restored from the checkpoint, and experiment.training.max_steps=100000 means continue training until Lightning reaches global step 100000.

  For weight-only initialization or finetuning without restoring optimizer/scheduler/global step, use:

  experiment.pretrained=/path/to/checkpoint.ckpt

  > Inconsistent dataset fallback on failed overrides:
  > When trying to aggressively override configuration nodes via the CLI to skip training (e.g., setting max_steps=0), Hydra silently switches the configuration context. For instance, while explicitly
  > requesting experiment=dino_wm_pusht, the script randomly attempted to look for the point_maze dataset paths, resulting in a FileNotFoundError for states.pth.

  This is not a random dataset fallback. In our Hydra setup, experiment and dataset are separate config groups.

  `experiment=dino_wm_pusht` selects the PushT training profile, but it does not automatically select the PushT dataset. If `dataset=...` is not specified, Hydra keeps the default dataset from `src/configs/
  config.yaml`, which is currently `dataset=dino_wm/point_maze`.

  So please specify both:
```
  python src/main.py experiment=dino_wm_pusht \
      dataset=dino_wm/pusht \
      model=nanowm_b2 \
      experiment.training.max_steps=100000
```
  
We updated the docs to use the fully qualified override paths, such as experiment.training.max_steps and experiment.resume_from_checkpoint, to avoid accidentally creating unused config keys with ++....

Again, thank you for your feedback, these have been very constructive. The related fixes and updated docs are now on main.
